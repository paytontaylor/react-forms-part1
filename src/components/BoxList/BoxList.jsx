import { useState } from "react";
import Box from "../Box/Box";
import NewBoxForm from "../NewBoxForm/NewBoxForm";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (box) => {
    setBoxes(boxes => [...boxes, box])
  }

  return (
    <div>
      <NewBoxForm addBox={addBox} />
      { boxes.map(({ id, height, width,color }) => <Box key={id} height={height} width={width} backgroundColor={color} />) }
    </div>
  )
}

export default BoxList;