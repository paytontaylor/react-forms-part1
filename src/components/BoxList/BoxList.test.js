import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import BoxList from "./BoxList";

it("should render BoxList without crashing", () => {
  render(<BoxList />)
})

it("should match snapshot", () => {
  const { asFragment } = render(<BoxList />)
  expect(asFragment()).toMatchSnapshot();
})