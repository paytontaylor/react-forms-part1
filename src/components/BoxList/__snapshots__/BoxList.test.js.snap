// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          id="height"
          name="height"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          id="width"
          name="width"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="color"
        >
          Color: 
        </label>
        <input
          id="color"
          name="color"
          type="text"
          value=""
        />
      </div>
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
